nvflare/fuel/utils/fobs/fobs.py

```diff
@@ -81,14 +143,28 @@ def unpack(self, obj: Any) -> Any:
 
         type_name = obj[FOBS_TYPE]
         if type_name not in _decomposers:
-            error = True
-            if _enum_auto_register:
-                cls = self._load_class(type_name)
-                if issubclass(cls, Enum):
-                    register_enum_types(cls)
-                    error = False
-            if error:
-                raise TypeError(f"Unknown type {type_name}, caused by mismatching decomposers")
+            registered = False
+            decomposer_name = obj.get(FOBS_DECOMPOSER)
+            cls = _load_class(type_name)
```

Comment:
What happens if this call raises exception? Should it be in try block?

Reply:
Added exception handling in _load_class() so only TypeError is raised.